test/DynamoCoreWpfTests/ViewExtensions/PythonMigrationViewExtensionTests.cs

```diff
@@ -0,0 +1,235 @@
+        public void WillDisplayDialogWhenOpeningGraphWithIronPythonNodes()
+        {
+            // Arrange
+            RaiseLoadedEvent(this.View);
+            var extensionManager = View.viewExtensionManager;
+            extensionManager.Add(viewExtension);
+
+            // Act
+            // open file
+            var examplePath = Path.Combine(UnitTestBase.TestDirectory, @"core\python", "python.dyn");
+            Open(examplePath);
+
+            var isIronPythonDialogOpen = this.View.OwnedWindows.Cast<Window>().Any(x=>x.GetType() == typeof(IronPythonInfoDialog));
+        
+            // Assert
+            Assert.IsTrue(isIronPythonDialogOpen);
+        }
+
+        /// <summary>
+        /// This test is created to check if a notification is logged to the Dynamo Logger 
+        /// when adding a using the IronPython engine
+        /// </summary>
+        [Test]
+        public void WillLogNotificationWhenAddingAnIronPythonNode()
+        {
+            // Arrange
+            string pythonNodeName = "Python Script";
+            RaiseLoadedEvent(this.View);
+            var raisedEvents = new List<string>();
+
+            // Act
+            // open file
+            this.ViewModel.Model.Logger.NotificationLogged += delegate (Dynamo.Logging.NotificationMessage obj)
+            {
+                raisedEvents.Add(obj.Sender);
+            };
+
+            var nodesCount = this.ViewModel.CurrentSpace.Nodes.Count();
+
+            this.ViewModel.ExecuteCommand(new DynamoModel.
+                CreateNodeCommand(Guid.NewGuid().ToString(), pythonNodeName, 0, 0, false, false));
+
+            // Assert
+            Assert.AreNotEqual(this.ViewModel.CurrentSpace.Nodes.Count(), nodesCount);
+            Assert.AreEqual(raisedEvents.Count, 1);
+            Assert.IsTrue(raisedEvents.Any(x => x.Contains(nameof(PythonMigrationViewExtension))));
+        }
+
+        /// <summary>
+        /// This test is created to check if a notification is logged to the Dynamo Logger 
+        /// only one time per open graph
+        /// </summary>
+        [Test]
+        public void WillOnlyLogNotificationWhenAddingAnIronPythonNodeOnce()
+        {
+            // Arrange
+            string pythonNodeName = "Python Script";
+            RaiseLoadedEvent(this.View);
+            var raisedEvents = new List<string>();
+
+            // Act
+            // open file
+            this.ViewModel.Model.Logger.NotificationLogged += delegate (Dynamo.Logging.NotificationMessage obj)
+            {
+                raisedEvents.Add(obj.Sender);
+            };
+
+            var nodesCount = this.ViewModel.CurrentSpace.Nodes.Count();
+
+            this.ViewModel.ExecuteCommand(new DynamoModel.
+                CreateNodeCommand(Guid.NewGuid().ToString(), pythonNodeName, 0, 0, false, false));
+            this.ViewModel.ExecuteCommand(new DynamoModel.
+                CreateNodeCommand(Guid.NewGuid().ToString(), pythonNodeName, 0, 0, false, false));
```

Comment:
i would also add an assertion that the added node has the `CPython` engine - otherwise we're not being explicit about this being the expected default behaviour